pkg/apb/client.go

```diff
@@ -175,20 +147,89 @@ func (c *Client) RunImage(
 	c.log.Debug("action:[ %s ]", action)
 	c.log.Debug("extra-vars:[ %s ]", extraVars)
 
+	// TODO(rhallisey): Cleanup the local vs incluster code paths so there
+	// is almost no variation between the two.
+	//
+	// If we're using a locally running broker we want to use the default
+	// namespace.
+	ns := context.Namespace
+	apbId := fmt.Sprintf("apb-%s", uuid.New())
+
+	sam := NewServiceAccountManager(c.log)
+	serviceAccountName, err := sam.CreateApbSandbox(ns, apbId)
+
+	if err != nil {
+		c.log.Error(err.Error())
+		return apbId, err
+	}
+
+	//TODO(rhallisey): The Kubernetes API objects should be in JSON structs
+	// so it's easier to manipulate them. This is for a rebase so we don't
+	// block the service account work.  I'll come back after and clean this
+	// up.
+	var pod *v1.Pod
 	if clusterConfig.InCluster {
-		return RunCommand("oc", "run", fmt.Sprintf("aa-%s", uuid.New()),
-			fmt.Sprintf("--image-pull-policy=Always"),
-			fmt.Sprintf("--image=%s", spec.Image), "--restart=Never",
-			"--", action, "--extra-vars", extraVars)
+		pod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: apbId,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "apb",
+						Image: spec.Image,
+						Args: []string{
+							action,
+							"--extra-vars",
+							extraVars,
+						},
+						ImagePullPolicy: v1.PullAlways,
+					},
+				},
+				RestartPolicy:      v1.RestartPolicyNever,
+				ServiceAccountName: serviceAccountName,
+			},
+		}
 	} else {
-		return RunCommand("oc", "run", fmt.Sprintf("aa-%s", uuid.New()),
-			"--env", fmt.Sprintf("OPENSHIFT_TARGET=%s", clusterConfig.Target),
-			"--env", fmt.Sprintf("OPENSHIFT_USER=%s", clusterConfig.User),
-			"--env", fmt.Sprintf("OPENSHIFT_PASS=%s", clusterConfig.Password),
-			fmt.Sprintf("--image-pull-policy=Always"),
-			fmt.Sprintf("--image=%s", spec.Image), "--restart=Never",
-			"--", action, "--extra-vars", extraVars)
+		// Set the namespace to default for the local broker.
+		ns = "default"
```

Comment:
branch for local runs is useful for testing with the broker when its not running inside a cluster, like for local development/testing. Ideally though, the branch would be minimal, it should only affect the the actual `oc login` call and definitely not impact namespace/etc.